### PR TITLE
Reportdownload

### PR DESCRIPTION
**Pull request by Anurodhyadav**
Basic Working GrammerBot API done.

### CODE REVIEW

File: src/views/Test.vue [hunk 1]
```diff
@@ -155,17 +160,60 @@ export default {
...
+          const encodedValue = encodeURIComponent(details[property]);
+          formBody.push(`${encodedKey}=${encodedValue}`);
+        }
+      }
+      formBody = formBody.join('&');
+      fetch('https://grammarbot.p.rapidapi.com/check', {
+        method: 'POST',
+        headers: {
+          'x-rapidapi-host': 'grammarbot.p.rapidapi.com',
+          'x-rapidapi-key': '5467d10b2bmsh8608f8e83c19152p1cecf1jsn649144d9df76',
```
Comment (Drishtantr) [inline src/views/Test.vue:182]: You probably have to store API keys in another file.

File: src/views/Test.vue [hunk 2]
```diff
@@ -155,17 +160,60 @@ export default {
...
-  },
-  created() {
-    const user = firebase.auth().currentUser;
-    this.current_User_email = user.email;
-    this.getAudio();
-  },
-  firestore: {
-    audioInfo: db.collection('audioInfo'),
-    userData: db.collection('UserData'),
+    Check() {
```
Comment (shakyaprasen) [inline src/views/Test.vue:163]: check should be in camelCase

File: src/views/Test.vue [hunk 3]
```diff
@@ -155,17 +160,60 @@ export default {
...
+      const details = {
+        language: 'en-US',
+        text: 'John and Bob ateing burgers every Friday night. Zack would love to hangs out and eat apples pie with his friend,but he has work a double shift tonight.',
+      };
+      let formBody = [];
+      // eslint-disable-next-line no-restricted-syntax
```
Comment (shakyaprasen) [inline src/views/Test.vue:169]: please don't use eslint-disable without any good reason.

File: src/views/Test.vue [hunk 4]
```diff
@@ -155,17 +160,60 @@ export default {
...
+      for (const property in details) {
```
Comment (shakyaprasen) [inline src/views/Test.vue:170]: you can use Object,keys(details).forEach here

File: src/views/Test.vue (hunk 1 above)
Comment (shakyaprasen) [reply src/views/Test.vue:182]: yes store api key in .env file

File: src/views/Test.vue [hunk 5]
```diff
@@ -155,17 +160,60 @@ export default {
...
+          'content-type': 'application/x-www-form-urlencoded;charset=UTF-8',
+        },
+        body: formBody,
+      })
+        .then(
+          (response) => response.json(),
+        )
+        .then((responseData) => {
+          const myCheck = (item) => {
+            console.log('Sentence:', item.sentence);
```
Comment (shakyaprasen) [inline src/views/Test.vue:192]: console.log still left here

**Review by shakyaprasen — APPROVED** [review]

File: src/views/Test.vue (hunk 4 above)
Comment (Anurodhyadav) [reply src/views/Test.vue:170]: Done

**Pull request merged**